lncore/addr.go

```diff
@@ -0,0 +1,319 @@
+package lncore
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"strconv"
+	"encoding/hex"
+
+	"github.com/mit-dci/lit/bech32"
+	"github.com/mit-dci/lit/crypto/fastsha256"
+)
+
+/*
+ * Ok so there's a lot of stuff going on here.  Lit natively supports addresses
+ * that are just the pkh of the pubkey encoded in bech32.  And that's pretty
+ * cool, you look up the IP+port in a tracker (or DHT) and connected with
+ * Noise_XX, then verify that the pubkey they provide you matches the pkh you
```

Comment:
So how lit should work after #437 is that if we haven't connected to the peer before, we still go ahead with NOISE_XX and get the pubkey of the host. But if we are connecting to a host whose pubkey we know already, we use NOISE_XK, since we already know the remote node's pubkey and theres no requirement to ask again. This is similar to how we using `ssh` asks you if you want to remember and I guess an implementing UI could ask the user for his preference as well